src/View/Form/ContextInterface.php

```diff
@@ -44,6 +44,15 @@ public function isPrimaryKey($field);
     public function isCreate();
 
     /**
+     * Sets the type of the form the context is created from.
+     *
+     * @param string $requestType The request type of the form the
+     *   context is created from.
+     * @return void
+     */
+    public function setRequestType($requestType);
```

Comment:
Couldn't we do something like

``php
$options['type'] = isset($options['type']) ? $options['type'] : null;
$context = $this->_getContext($options['context'], $options['type']);
$isCreate = $context->isCreate();
```

The context constructor could use the provided type or fallback to the result of `$this->isCreate()`.